File: php/manage_saved_searches.php
<!DOCTYPE html>
<html>
<head>
  <meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
</head>
<body>
<?php
// tmhOAuthを初期化
require "./tmhOAuth.php";
$twitter = new tmhOAuth(
  // 実際に取得したコンシューマキー、アクセストークンを記述
  array("consumer_key" => "コンシューマキー",
    "consumer_secret" => "コンシューマシークレット",
    "user_token" => "アクセストークン",
    "user_secret" => "アクセストークンシークレット"));

if (isset($_REQUEST["command"])) {
  switch($_REQUEST["command"]){
    case "create":
      // 保存した検索を作成する
      $query = $_REQUEST["query"];
      $twitter->request("POST", $twitter->url("1/saved_searches/create")
        , array("query" => $query));
      print "新しい保存した検索: " . htmlspecialchars($query) . " を作成しました<br>";
      break;
    case "delete":
      // 指定した保存した検索を削除する
      $twitter->request("DELETE", $twitter->url("1/saved_searches/destroy/" . $_REQUEST["id"]));
      print "保存した検索を削除しました<br>";
      break;
  }
  ?><br>
  <?php
}

// スクリーン名を取得
$twitter->request("GET", $twitter->url("1/account/verify_credentials"));
$screenName = json_decode($twitter->response["response"])->{"screen_name"};
print "@" . $screenName . "の保存した検索一覧:<br>";

// 保存した検索一覧を取得
$twitter->request("GET", $twitter->url("1/saved_searches"));
$savedSearches = json_decode($twitter->response["response"]);
foreach ($savedSearches as $savedSearch) {
  print "<a href='./manage_saved_searches.php?command=delete&id=" . $savedSearch->{"id_str"} . "'>削除</a> "
    . htmlspecialchars($savedSearch->{"query"}) . "<br>";
}
?>
<form action="./manage_saved_searches.php" method="POST">
  <input type="hidden" name="command" value="create"/>
  保存する検索:<input type="text" name="query"/><br>
  <input type="submit" value="保存した検索を作成"/>
</form>

</body>
</html>

File: java/src/main/java/ManageSavedSearches.java
import twitter4j.*;
import twitter4j.auth.AccessToken;

import java.util.List;

public class ManageSavedSearches {
  public static void main(String[] args) {
    // 使用法: java ManageSavedSearches [コマンド create / delete] [新規作成するクエリ / 削除するid]
    Twitter twitter = new TwitterFactory().getInstance();
    // 実際に取得したコンシューマキー、アクセストークンを記述
    String consumerKey = "コンシューマキー";
    String consumerSecret = "コンシューマシークレット";
    String accessToken = "アクセストークン";
    String accessTokenSecret = "アクセストークンシークレット";

    // OAuth トークンを設定
    twitter.setOAuthConsumer(consumerKey, consumerSecret);
    twitter.setOAuthAccessToken(new AccessToken(accessToken, accessTokenSecret));
    try {
      if (args.length == 2) {
        String command = args[0];
        if ("create".equals(command)) {
          String query = args[1];
          twitter.createSavedSearch(query);
          System.out.println("新しい保存した検索 " + query + " を作成しました");
        } else if ("delete".equals(command)) {
          int id = Integer.parseInt(args[1]);
          twitter.destroySavedSearch(id);
          System.out.println("保存した検索を削除しました");
        }
      }
      System.out.println("@" + twitter.getScreenName() + "の保存した検索一覧:");
      List<SavedSearch> savedSearches = twitter.getSavedSearches();
      for (SavedSearch savedSearch : savedSearches) {
        System.out.println(savedSearch.getQuery() + " (id:" + savedSearch.getId() + ")");
      }
    } catch (TwitterException te) {
      System.out.println("API呼び出しに失敗しました: " + te.getMessage());
    }
  }
}
